src/store/plugins/sdk13.js

```diff
@@ -0,0 +1,223 @@
+
+export default (store) => {
+  let sdk;
+  const aeppInfo = {};
+
+  store.registerModule('sdk13Plugin', {
+    namespaced: true,
+    state: {
+      ready: false,
+    },
+    getters: {
+      sdk: ({ ready }) => (ready ? sdk : undefined),
+    },
+    mutations: {
+      setSdkReady(state) {
+        state.ready = true;
+      },
+    },
+    actions: {
+      async initialize({ commit }) {
+        if (sdk) {
+          return;
+        }
+        await watchUntilTruthy(
+          () => store.state.isRestored && !isEmpty(store.getters.account),
+        );
+
+        const { activeNetwork } = store.getters;
+
+        // TODO: use this method for `onMessageSign` replacement.
+        // eslint-disable-next-line no-unused-vars
+        const signCb = async (payload, options, origin = null) => {
+          try {
+            const host = origin ? new URL(origin).host : null;
+            const permission = (!host && IS_CORDOVA) || await store.dispatch('permissions/checkPermissions', {
+              host,
+              method: 'sign',
+              params: payload,
+            });
+
+            return store.dispatch('accounts/signTransaction', {
+              txBase64: payload,
+              opt: {
+                ...options,
+                modal: !permission,
+                host,
+              },
+            });
+          } catch (error) {
+            throw new RpcRejectedByUserError(error.message);
+          }
+        };
+
+        const signCbBackground = async (method, payload, options) => {
+          const aepp = aeppInfo[options.aeppRpcClientId];
+          const isTxSigning = method === 'transaction.sign';
+          if (
+            (await store.dispatch('permissions/checkPermissions', {
+              host: new URL(aepp.origin).host,
+              method,
+              params: options,
+            }))
+            || (await showPopup(
+              aepp.origin,
+              isTxSigning ? POPUP_TYPE_SIGN : POPUP_TYPE_MESSAGE_SIGN,
+              {
+                ...options,
+                ...(isTxSigning)
+                  ? {
+                    tx: payload,
+                    txObject: unpackTx(payload),
+                  }
+                  : {
+                    message: payload,
+                  },
+              },
+            ).then(
+              () => true,
+              () => false,
+            ))
+          ) {
+            if (method === 'message.sign') {
+              return store.dispatch('accounts/sign', messageToHash(payload));
+            }
+            return store.dispatch('accounts/signTransaction', {
+              txBase64: payload,
+              opt: {
+                ...options,
+                modal: false,
+                host: aepp.origin,
+              },
+            });
+          }
+          throw new RpcRejectedByUserError('Rejected by user');
+        };
+
+        // TODO: move this to a seperate file with typescript and combine the custom type
```

Comment:
Why not just do it now? :)

Reply:
But this class is new code - it's not refactor.

Reply:
My comment was referring only to the `SuperheroSdk` class, so it's around of 20 lines. I'm not forcing you to rewrite whole file.
The reason I'd like this code to be moved into separate TS file is that after checking it TS is complaining about this piece of code. And I really don't want us to add new problematic code into repository. There is already too much of "temporary hacks".

Reply:
![image](https://user-images.githubusercontent.com/8217670/229117765-7495ef06-f5fe-4c25-a1d9-d6052f60e3e9.png)

Reply:
Define appropriate files.